swap_meet/vendor.py

```diff
@@ -1,2 +1,136 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    '''
+    Instances of Vendor Class represent a vendor at a swap
+    meet. Their inventory attribute stores a list of
+    instances of Item classes, representing the items
+    they are bringing to the swapmeet.
+    '''
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, new_item):
+        '''
+        Add an item to a vendor's inventory list.
+
+        Parameters - 
+            Item to be added.
+        Returns -
+            Item that was added.
+        '''
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item):
+
+        '''
+        Remove an item from a vendor's inventory list.
+        
+        Parameters - 
+            Item to be removed.
+        Returns -
+            Item that was removed.
+        '''
+
+        try:
+            self.inventory.remove(item)
+        except(ValueError):
+            return False
+        return item
+
+    def get_by_category(self, category_to_search):
+
+        '''
+        Parameters - 
+            Desired category to be searched
+
+        Returns - 
+            List of items in vendor's inventory matching
+            the desired categpry.
+        '''
+
+        return_list = []
+        for item in self.inventory:
+            if item.category == category_to_search:
+                return_list.append(item)
+        return return_list
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        '''
+        Swaps two Vendors' items and updates their inventories to reflect the swap.
+
+        Parameters - 
+            Other Vendor, Item to be swapped from self, Item to be swapped from the other vendor.
+
+        Returns -
+            Boolean based on if the items are present in the correct inventories.
+
+        '''
+
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        self.add(other_vendor.remove(their_item))
+        other_vendor.add(self.remove(my_item))
```

Comment:
I think I misled you with this implementation from viewing-party. My mistake! I thought `remove` returned the item like `pop` does. :(

So what is actually being appended is a `None` value rather than the actual item. We'll need to append and remove individually